src/containers/App/index.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+import type {AppProps, AppState} from './types';
+import React, {Component} from 'react';
+import {Route} from 'react-router-dom';
+import {connect} from 'react-redux';
+import './App.css';
+import {Logger} from '../../services/logger';
+import Home from '../Home';
+import AppHeader from '../../components/AppHeader';
+import AppFooter from '../../components/AppFooter';
+
+export class App extends Component<AppProps, AppState> {
+  componentDidMount() {
```

Comment:
this should be a pure component, componentDidMount is doing nothing asynchronous such as a dispatch to an async function.
 
This component could be rewritten as:

```javascript
// @flow
import type {AppProps, AppState} from './types';
import React, {Component} from 'react';
import {Route} from 'react-router-dom';
import {connect} from 'react-redux';
import './App.css';
import {Logger} from '../../services/logger';
import Home from '../Home';
import AppHeader from '../../components/AppHeader';
import AppFooter from '../../components/AppFooter';

const App = ({ buildInformation: BuildInformationType }) => {
    const logger = new Logger();
    logger.log('testing', 'Info');
    return (
      <div className="App">
        <AppHeader />
        <Route exact path="/" component={Home} />
        <AppFooter buildInformation={buildInformation} />
      </div>
    );
}

const mapStateToProps = ({ app }: TypeofApp): TypeOfProps => ({
 buildInformation: app.buildInformation
})

export default connect(mapStateToProps)(App);
```

Reply:
Same as Home. This is the main component of the app, it will probably need a state when you start adding functionality to it...

Reply:
@ferenckv functionality not always requires an internal state, especially if we have redux which will handle the global one and all data is manage on props.

You can't leave components in an invalid state by assuming they will need something else in the future.

Reply:
Also, in this case, the goal is to log when the app is loaded, not every time the component is rendered.